odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
```

Comment:
This is already a bit smarter than the old copy-paste solution. But I agree, this still feels a bit clumsy.

Static method sounds good.